drivers/adc/adc_stm32.c

```diff
@@ -567,8 +567,8 @@ static int adc_stm32_channel_setup(const struct device *dev,
 		return -EINVAL;
 	}
 
-	if (channel_cfg->channel_id == ADC_CHANNEL_TEMPSENSOR ||
-		channel_cfg->channel_id == ADC_CHANNEL_VREFINT) {
+	if ((channel_cfg->channel_id == (uint8_t) ADC_CHANNEL_TEMPSENSOR) ||
```

Comment:
Are you sure ADC_CHANNEL_TEMPSENSOR cast be cast on uint8_t without loosing information? I don't think so.
I think the problem is not just coverity, but there is a real bug behind.
Probably ` __LL_ADC_DECIMAL_NB_TO_CHANNEL() ` could be used on `channel_cfg->channel_id` to be able to compare it to ADC_CHANNEL_TEMPSENSOR.

Reply:
Yes, so in order to compare apples to apples, I propose the following comparison :
`__LL_ADC_DECIMAL_NB_TO_CHANNEL(channel_cfg->channel_id) ==  ADC_CHANNEL_TEMPSENSOR`

Reply:
So I will try this and update you soon.

Reply:
I have tested `samples/sensor/stm32_temp_sensor` in nucleo_f103rb board. and it's working.